appinventor/components/src/com/google/appinventor/components/runtime/Clock.java

```diff
@@ -207,6 +216,102 @@ public static Calendar MakeInstant(String from) {
+   * @param month month integer
+   * @param day day integer
+   * @return  Calendar instant
+   */
+  @SimpleFunction(description = "Allows the user to set the clock to be "
+    +"a date value.\n" +
+    "Valid values for the month field are 1-12 and 1-31 for the day field.\n")
+  public Calendar MakeDate(int year, int month, int day) {
+    int jMonth = month - 1;
+    try {
+      GregorianCalendar cal = new GregorianCalendar(year, jMonth, day);
+      cal.setLenient(false);
+      cal.getTime();
+    } catch (java.lang.IllegalArgumentException e) {
+      form.dispatchErrorOccurredEvent(this, "ClockMakeDate", ErrorMessages.ERROR_ILLEGAL_DATE);
+    }
+    
+    Calendar instant = Dates.DateInstant(year, month, day);
+    return instant;
+  }
+
+  /**
+   * An instant in time specified by hh:mm
+   * where hh is the hour, mm the minute
+   * @param hour hour integer
+   * @param minute minute integer
+   * @return  Calendar instant since 1/1/1970
+   */
+  @SimpleFunction(description = "Allows the user to set the time of the clock - " +
+    "Valid format is hh:mm\n")
+  public Calendar MakeTime(int hour, int minute) {
+    Calendar instant = null;
+    try {
+      String time_str = String.valueOf(hour) + ":" + String.valueOf(minute);
+      DateFormat timeFormat = new SimpleDateFormat("hh:mm");
+      timeFormat.setLenient(false);
+      Date time = timeFormat.parse(time_str);
+      instant = Calendar.getInstance();
+      instant.setTime(time);
+    }catch (Exception e) {
+    throw new YailRuntimeError(
+        "Argument to ClockMakeTime should have form hh:mm",
+        "Sorry to be so picky.");
+    }
+
+    
+    return instant;
+  }
+
+  /**
+   * An instant in time specified by MM/DD/YYYY hh:mm
+   * where MM is the month (01-12), DD the day (01-31), YYYY the year
+   * hh is the hour, mm the minute
+   * @param year year integer
+   * @param month month integer
+   * @param day day integer
+   * @param hour hour integer
+   * @param minute minute integer
+   * @return  Calendar instant
+   */
+  @SimpleFunction(
+    description = "Allows the user to set the date and time to be displayed when the clock opens.\n" +
+    "Valid values for the month field are 1-12 and 1-31 for the day field.\n")
+  public Calendar MakeInstantFromParts(int year, int month, int day, int hour, int minute, int second) {
+    int jMonth = month - 1;
+    Calendar instant = null;
+    try {
+      instant = new GregorianCalendar(year, jMonth, day);
+      instant.setLenient(false);
+      instant.getTime();
+    } catch (java.lang.IllegalArgumentException e) {
+      form.dispatchErrorOccurredEvent(this, "MakeInstantFromParts", ErrorMessages.ERROR_ILLEGAL_DATE);
+    }
+    
+    instant = Dates.DateInstant(year, month, day);
+    
+    try {
+      instant.set(Calendar.HOUR_OF_DAY, hour);
+      instant.set(Calendar.MINUTE, minute);
+      instant.set(Calendar.SECOND, second);
+
+    }catch (Exception e) {
+    throw new YailRuntimeError(
```

Comment:
Indentation plus motivation for throwing YailRuntimeError? I think I prefer using dispatchErrorOccurredEvent.